docs/fsharp/style-guide/component-design-guidelines.md

```diff
@@ -693,11 +693,83 @@ let checkNonNull argName (arg: obj) =
     | null -> nullArg argName
     | _ -> ()
 
-let checkNonNull` argName (arg: obj) =
+let checkNonNull' argName (arg: obj) =
     if isNull arg then nullArg argName
     else ()
 ```
 
+Starting with F# 9, you can leverage the new `| null` [syntax](../language-reference/values/null-values.md#null-values-starting-with-f-9) to make the compiler indicate possible null values and where they need handling. The above code will then start producing warnings:
+
+```fsharp
+let checkNonNull argName (arg: obj) =
+    match arg with
+    // nullness warning: the type 'obj' does not support 'null'
+    | null -> nullArg argName
+    | _ -> ()
+
+let checkNonNull' argName (arg: obj) =
+    // nullness warning: the type 'obj' does not support 'null'
+    if isNull arg then nullArg argName 
+    else ()
+```
+
+To address the warnings, you could explicitly specify `null` as a possible argument value:
```

Comment:
This needs rework based on how the first example is redone.
Right now, it shows the wrong incentive.

If e.g. the first example shows a warning for (string|null).Length, that this example should show how to handle it.

Reply:
I removed and reworked the text, take a look if it's more consistent and helpful now.